test/e2e/hot-reload.test.js

```diff
@@ -25,6 +25,19 @@ describe('hot-reload', () => {
     let originalGatewayPort = null;
     let watcher = null;
 
+    function startServer (options) {
+      return new Promise(
+        (resolve, reject) => {
+          childProcess = spawn('node', [options.modulePath], {cwd: options.cwd, env: options.env});
+          childProcess.stdout.on('data', (data) => {
+            const chk = /gateway http server listening on/g;
+            if (chk.test(data)) {
+              resolve('server started');
```

Comment:
I'd just resolve the promise, with no returned data.

Reply:
Also — maybe we shuold reject the promise if the reg exp does not pass, so we do not waste our time waiting for a resolve that'll never happen?